packages/dashboard/src/hooks/useCreateTaskForm.tsx

```diff
@@ -42,9 +43,21 @@ export const useCreateTaskFormData = (rmf: RmfIngress | undefined) => {
         setWaypointNames(waypointNames);
       }),
     );
+    subs.push(
+      rmf.fleetsObs.subscribe((fleetStates) => {
+        const result: Record<string, string[]> = {};
+        for (const fleet of fleetStates) {
+          if (!fleet.name || !fleet.robots) {
+            continue;
+          }
+          result[fleet.name] = Object.keys(fleet.robots);
+        }
+        setFleets(result);
```

Comment:
This is currently used within a single `useCreateTaskFormData` call when `appbar` is loaded, so it only ever triggers a render once. This can be verified with running the hotel world.

Reply:
Tried this out, so it doesn't re-render because the api-client has a "fake" fleet observable, it only ever emits once and doesn't update when the fleet changes. Not sure if that is a bug but it's become a "feature" now, so probably just add a comment at api-client.

https://github.com/open-rmf/rmf-web/blob/d9c8fbabe1c7f08bf4ba614bcd7bcfeb0d294e23/packages/dashboard/src/services/rmf-ingress.ts#L226-L231

Reply:
Good catch, thanks for pointing it out. https://github.com/open-rmf/rmf-web/pull/1004/commits/7ba206421c5899f57c5c38f92e8c96e8fe785a7f